### PR TITLE
Add possibility to use default fiori styles in all renderers

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- slight refactor of gulpfile to make it more readable
- add styles.(s)css to markdown-renderer so that anyone can import them to use nicer looking defaults

**Related issue(s)**

See also https://github.com/kyma-project/console/issues/684
Resolves https://github.com/kyma-incubator/documentation-component/issues/33

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -96,7 +128,10 @@ rollupModules.forEach(mod => {
     const packageName = path.resolve(__dirname, `${sources}/${mod}`);
 
     await buildByRollup(packageName);
-    await fse.copy(`${packageName}/lib`, `${dist}/${packageNames[mod]}/lib`);
+    if (
+      `${packageName}/lib` !== `${__dirname}/${dist}/${packageNames[mod]}/lib`
+    )
```
Comment (magicmatatjahu) [inline gulpfile.js]: Please add semicolons.

File: gulpfile.js [hunk 2]
```diff
@@ -65,20 +75,42 @@ const rollupPackages = {
...
 
-rollupModules.concat(modules).forEach(mod => {
-  gulp.task(`${mod}:install`, async () => {
-    const packageName = path.resolve(__dirname, `${sources}/${mod}`);
-    await install(packageName);
-  });
-});
+const install = async () => {
+  return Promise.all(
+    allModules.map(mod => {
```
Comment (magicmatatjahu) [inline gulpfile.js]: `allModules.map(mod => installPackage(path.resolve(__dirname, `${sources}/${mod}`)))`

Comment (aerfio) [reply gulpfile.js]: existing solution is more readable, those gulpfiles get really nasty very fast so I'd rather be more verbose

File: packages/open-api-render-engine/README.md [hunk 3]
```diff
@@ -7,20 +7,30 @@ The OpenAPI render engine is based on the [Swagger UI](https://github.com/swagge
...
+| ----------- | -------- | ----- | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `plugins`   | No       | `any` | `[]`          | Defines custom plugins for the [Swagger UI](https://github.com/swagger-api/swagger-ui) component. Read [this](https://github.com/swagger-api/swagger-ui/blob/master/docs/customization/plugin-api.md) document for more details. |
+| `schemaUrl` | No       | `any` | `{}`          | Defines the custom schema for the [SwaggerUI](https://github.com/swagger-api/swagger-ui) component.                                                                                                                              |
+
+> **NOTE**: To learn how to pass options of a render engine, read [this](../../docs/props/render-engines.md#pass-global-options) document.
+
+## Styles
```
Comment (magicmatatjahu) [inline packages/open-api-render-engine/README.md]: Please also update https://github.com/kyma-incubator/documentation-component/blob/master/docs/templates/document-templates/render-engine_README.md

File: gulpfile.ts [hunk 4]
```diff
@@ -122,83 +158,67 @@ gulp.task(
 );
 
 gulp.task("build:normal", gulp.series(modules));
-gulp.task("build", gulp.series("build:rollup", "build:normal"));
+
+const copyOpenApiStyleAssets = () =>
+  gulp
+    .src(`${path.resolve(sources, openApiRenderEngine)}/src/assets/*`)
+
```
Comment (magicmatatjahu) [inline gulpfile.ts]: Please remove blank line.

File: packages/async-api-render-engine/package.json [hunk 5]
```diff
@@ -29,7 +29,8 @@
     "@kyma-project/asyncapi-react": "0.2.0"
   },
   "peerDependencies": {
-    "@kyma-project/documentation-component": "^0.0.1"
+    "@kyma-project/documentation-component": "^0.2.0-alpha",
+    "fiori-fundamentals": "^1.4.3"
```
Comment (magicmatatjahu) [inline packages/async-api-render-engine/package.json]: Please remove it.

File: packages/markdown-render-engine/README.md [hunk 6]
```diff
@@ -30,6 +32,15 @@ The Markdown render engine is based on the [React Markdown](https://github.com/r
 
 > **NOTE**: To learn how to pass options of a render engine, read [this](../../docs/props/render-engines.md#pass-global-options) document.
 
+### Styles
```
Comment (magicmatatjahu) [inline packages/markdown-render-engine/README.md]: Please only mention about our styling inside package and add additional info in `blockquote` block about fonts from `fiori`.

File: packages/markdown-render-engine/package.json [hunk 7]
```diff
@@ -43,7 +43,8 @@
     "@types/react-copy-to-clipboard": "^4.2.6"
   },
   "peerDependencies": {
-    "@kyma-project/documentation-component": "^0.0.1"
+    "@kyma-project/documentation-component": "^0.0.1",
+    "fiori-fundamentals": "^1.4.3"
```
Comment (magicmatatjahu) [inline packages/markdown-render-engine/package.json]: Remove this package

File: packages/markdown-render-engine/src/styles.scss [hunk 8]
```diff
@@ -0,0 +1,367 @@
+%tabsStyling {
+  .cms__tabs {
+    box-sizing: border-box;
+    width: 100%;
+    height: 100%;
+    font-family: "72";
```
Comment (magicmatatjahu) [inline packages/markdown-render-engine/src/styles.scss]: Add fallback to normal fonts.

Comment (magicmatatjahu) [reply packages/markdown-render-engine/src/styles.scss]: Please also do it in another places in this file.

File: packages/open-api-render-engine/README.md [hunk 9]
```diff
@@ -7,20 +7,31 @@ The OpenAPI render engine is based on the [Swagger UI](https://github.com/swagge
...
+| ----------- | -------- | ----- | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `plugins`   | No       | `any` | `[]`          | Defines custom plugins for the [Swagger UI](https://github.com/swagger-api/swagger-ui) component. Read [this](https://github.com/swagger-api/swagger-ui/blob/master/docs/customization/plugin-api.md) document for more details. |
+| `schemaUrl` | No       | `any` | `{}`          | Defines the custom schema for the [SwaggerUI](https://github.com/swagger-api/swagger-ui) component.                                                                                                                              |
+
+> **NOTE**: To learn how to pass options of a render engine, read [this](../../docs/props/render-engines.md#pass-global-options) document.
+
+### Styles
```
Comment (magicmatatjahu) [inline packages/open-api-render-engine/README.md]: Please only mention about our styling inside package and add additional info in blockquote block about fonts from fiori.

Comment (magicmatatjahu) [reply packages/open-api-render-engine/README.md]: Please also write about default styling from `swagger-ui` and add appropriate import for that.

File: packages/open-api-render-engine/package.json [hunk 10]
```diff
@@ -33,7 +33,8 @@
     "@types/swagger-ui-dist": "^3.0.0"
   },
   "peerDependencies": {
-    "@kyma-project/documentation-component": "^0.0.1"
+    "@kyma-project/documentation-component": "^0.2.0-alpha",
+    "fiori-fundamentals": "^1.4.3"
```
Comment (magicmatatjahu) [inline packages/open-api-render-engine/package.json]: Please remove it.

File: packages/open-api-render-engine/src/renderEngine.tsx [hunk 11]
```diff
@@ -1,6 +1,7 @@
 import React, { useEffect } from "react";
 import { RenderEngineProps } from "@kyma-project/documentation-component";
 import { OpenApiProps } from "./types";
+import "swagger-ui-dist/swagger-ui.css";
```
Comment (magicmatatjahu) [inline packages/open-api-render-engine/src/renderEngine.tsx]: Remove it.

File: playground/package.json [hunk 12]
```diff
@@ -13,35 +12,20 @@
...
-  },
-  "devDependencies": {
-    "@types/node": "^10.12.24",
-    "@types/react": "^16.8.22",
-    "@types/react-dom": "^16.8.4",
-    "@types/styled-components": "^4.1.16",
-    "@types/react-sticky": "^6.0.3",
-    "customize-cra": "^0.2.14",
+    "fiori-fundamentals": "^1.7.1",
+    "node-sass": "^4.12.0",
```
Comment (magicmatatjahu) [inline playground/package.json]: You have node-sass in root so please remove it

File: packages/open-api-render-engine/src/renderEngine.tsx (hunk 11 above)
Comment (aerfio) [reply packages/open-api-render-engine/src/renderEngine.tsx]: why would I do it, swagger component is literally unusable without default styles

File: packages/open-api-render-engine/README.md (hunk 9 above)
Comment (aerfio) [reply packages/open-api-render-engine/README.md]: Swagger-ui is unusable without those defaults, so I think we shouldn't even consider leaving it as a choice for a user

File: packages/markdown-render-engine/README.md [hunk 13]
```diff
@@ -7,51 +7,67 @@ The Markdown render engine is based on the [React Markdown](https://github.com/r
...
+| `headingPrefix`   | No       | `string or ((source: Source) => string)`, where [`Source`](../documentation-component/src/interfaces/Source.ts) is type | `""`          | Defines a prefix for any heading in a Markdown file.                                                                                                                                                                                                                                 |
+| `copyButton`      | No       | `ReactNode`                                                                                                             | `null`        | Defines a custom copy button in code blocks. The Markdown render engine accepts only code blocks as a value to copy.                                                                                                                                                                 |
 
 > **NOTE**: To learn how to pass options of a render engine, read [this](../../docs/props/render-engines.md#pass-global-options) document.
 
+### Styles
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/) import them as follows:
```
Comment (kazydek) [inline packages/markdown-render-engine/README.md]:
```suggestion
To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), import them as follows:
```

File: packages/markdown-render-engine/README.md [hunk 14]
```diff
@@ -7,51 +7,67 @@ The Markdown render engine is based on the [React Markdown](https://github.com/r
...
+
+```js
+import "@kyma-project/dc-markdown-render-engine/lib/styles.css";
+```
+
+To use default font named `72` provided by SAP Fundamentals follow its [Getting Started](https://sap.github.io/fundamental/getting-started.html) guide and import it like follows:
```
Comment (kazydek) [inline packages/markdown-render-engine/README.md]:
```suggestion
To use the default font `72` provided by SAP Fundamentals, install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the font as follows:
```

File: packages/odata-render-engine/README.md [hunk 15]
```diff
@@ -7,16 +7,27 @@ The OData render engine is based on the [OData React](https://github.com/kyma-in
...
 
 ## Prerequisites
 
-- [`odata-react`](https://github.com/kyma-incubator/documentation-component/tree/master/packages/odata-react): ^0.0.1
+- [`odata-react`](https://github.com/kyma-incubator/documentation-component/tree/master/packages/odata-react): ^0.2.0-alpha.2
 - [`styled-components`](https://github.com/styled-components/styled-components): >= 3.4.0
+
+### Styles
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/) follow its [Getting Started](https://sap.github.io/fundamental/getting-started.html) guide and import it like follows:
```
Comment (kazydek) [inline packages/odata-render-engine/README.md]:
```suggestion
To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the styles as follows:
```

File: packages/open-api-render-engine/README.md [hunk 16]
```diff
@@ -7,20 +7,36 @@ The OpenAPI render engine is based on the [Swagger UI](https://github.com/swagge
...
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/) import them as follows:
+
+```js
+import "@kyma-project/dc-markdown-render-engine/lib/styles.css";
+```
+
+To use default font named `72` provided by SAP Fundamentals follow its [Getting Started](https://sap.github.io/fundamental/getting-started.html) guide and import it like follows:
```
Comment (kazydek) [inline packages/open-api-render-engine/README.md]:
```suggestion
To use the default font `72` provided by SAP Fundamentals, install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the font as follows:
```

File: packages/markdown-render-engine/README.md [hunk 17]
```diff
@@ -7,51 +7,67 @@ The Markdown render engine is based on the [React Markdown](https://github.com/r
...
 
+### Styles
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), import them as follows:
+
+```js
+import "@kyma-project/dc-markdown-render-engine/lib/styles.css";
+```
+
+To use the default font `72` provided by SAP Fundamentals, install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the font as follows:
```
Comment (kazydek) [inline packages/markdown-render-engine/README.md]:
```suggestion
To use the default font `72` provided by SAP Fundamentals, install the npm package from [this](https://sap.github.io/fundamental/getting-started.html) Getting Started guide and import the font as follows:
```

File: packages/odata-render-engine/README.md [hunk 18]
```diff
@@ -7,16 +7,27 @@ The OData render engine is based on the [OData React](https://github.com/kyma-in
...
 
 ## Prerequisites
 
-- [`odata-react`](https://github.com/kyma-incubator/documentation-component/tree/master/packages/odata-react): ^0.0.1
+- [`odata-react`](https://github.com/kyma-incubator/documentation-component/tree/master/packages/odata-react): ^0.2.0-alpha.2
 - [`styled-components`](https://github.com/styled-components/styled-components): >= 3.4.0
+
+### Styles
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the styles as follows:
```
Comment (kazydek) [inline packages/odata-render-engine/README.md]:
```suggestion
To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), install the npm package from [this](https://sap.github.io/fundamental/getting-started.html) Getting Started guide and import the styles as follows:
```

File: packages/open-api-render-engine/README.md [hunk 19]
```diff
@@ -7,20 +7,36 @@ The OpenAPI render engine is based on the [Swagger UI](https://github.com/swagge
...
+| ----------- | -------- | ----- | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `plugins`   | No       | `any` | `[]`          | Defines custom plugins for the [Swagger UI](https://github.com/swagger-api/swagger-ui) component. Read [this](https://github.com/swagger-api/swagger-ui/blob/master/docs/customization/plugin-api.md) document for more details. |
+| `schemaUrl` | No       | `any` | `{}`          | Defines the custom schema for the [SwaggerUI](https://github.com/swagger-api/swagger-ui) component.                                                                                                                              |
+
+> **NOTE**: To learn how to pass options of a render engine, read [this](../../docs/props/render-engines.md#pass-global-options) document.
+
+### Styles
+
+To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/) import them as follows:
```
Comment (kazydek) [inline packages/open-api-render-engine/README.md]:
```suggestion
To use default styles based on [SAP Fundamentals](https://sap.github.io/fundamental/), import them as follows:
```

File: packages/open-api-render-engine/README.md [hunk 20]
```diff
@@ -7,20 +7,36 @@ The OpenAPI render engine is based on the [Swagger UI](https://github.com/swagge
...
+
+```js
+import "@kyma-project/dc-markdown-render-engine/lib/styles.css";
+```
+
+To use the default font `72` provided by SAP Fundamentals, install [this](https://sap.github.io/fundamental/getting-started.html) npm package and import the font as follows:
```
Comment (kazydek) [inline packages/open-api-render-engine/README.md]:
```suggestion
To use the default font `72` provided by SAP Fundamentals, install the npm package from [this](https://sap.github.io/fundamental/getting-started.html) Getting Started guide and import the font as follows:
```

File: packages/open-api-render-engine/src/renderEngine.tsx (hunk 11 above)
Comment (magicmatatjahu) [reply packages/open-api-render-engine/src/renderEngine.tsx]: But then you must overwrite whole styles if you want add yours.

**Pull request merged**